### PR TITLE
Plugins: Move PinnedPlugins component

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Related #6115.

This is pre-requisite to offer `PluginModal` component inside `@wordpress/editor` package. It is page agnostic so I believe that it shouldn't be tied to the `edit-post` module the same way `PluginSidebar` is.

To make it possible this PR move `PinnedPlugins` component to `@wordpress/plugins` package to allow reuse between different packages. This include also related data store. I think that the feature of pinning should work the same regardless of which page you open - whether it is going to be the edit post page or edit template page or whatever the future brings.

## How has this been tested?
I copied the following plugin code into JS console to ensure pinning feature still works:

```js
var Button = wp.components.Button;
var PanelBody = wp.components.PanelBody;
var PanelRow = wp.components.PanelRow;
var compose = wp.compose.compose;
var withDispatch = wp.data.withDispatch;
var withSelect = wp.data.withSelect;
var PlainText = wp.editor.PlainText;
var Fragment = wp.element.Fragment;
var el = wp.element.createElement;
var __ = wp.i18n.__;
var registerPlugin = wp.plugins.registerPlugin;
var PluginSidebar = wp.editPost.PluginSidebar;
var PluginSidebarMoreMenuItem = wp.editPost.PluginSidebarMoreMenuItem;

function SidebarContents( props ) {
	return el(
		PanelBody,
		{},
		el(
			PanelRow,
			{},
			el(
				'label',
				{
					'htmlFor': 'title-plain-text'
				},
				__( 'Title:' ),
			),
			el(
				PlainText,
				{
					id: 'title-plain-text',
					onChange: props.updateTitle,
					placeholder: __( '(no title)' ),
					value: props.title
				}
			)
		),
		el(
			PanelRow,
			{},
			el(
				Button,
				{
					isPrimary: true,
					onClick: props.resetTitle
				},
				__( 'Reset' )
			)
		)
	);
}

var SidebarContentsWithDataHandling = compose( [
	withSelect( function( select ) {
		return {
			title: select( 'core/editor' ).getEditedPostAttribute( 'title' ),
		};
	} ),
	withDispatch( function( dispatch ) {
		function editPost( title ) {
			dispatch( 'core/editor' ).editPost( {
				title: title
			} );
		}

		return {
			updateTitle: function( title ) {
				editPost( title );
			},
			resetTitle: function() {
				editPost( '' );
			}
		};
	} )
] )( SidebarContents );

function MySidebarPlugin() {
	return el(
		Fragment,
		{},
		el(
			PluginSidebar,
			{
				name: 'title-sidebar',
				title: __( 'Sidebar title plugin' )
			},
			el(
				SidebarContentsWithDataHandling,
				{}
			)
		),
		el(
			PluginSidebarMoreMenuItem,
			{
				target: 'title-sidebar'
			},
			__( 'Sidebar title plugin' )
		)
	);
}

registerPlugin( 'my-sidebar-plugin', {
	icon: 'text',
	render: MySidebarPlugin
} );
```

## Types of changes
Refactoring

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -202,7 +202,7 @@
 		"*.js": [
 			"eslint"
 		],
-		"{docs/{root-manifest.json,tool/*.js},packages/{*/README.md,*/src/{actions,selectors}.js,components/src/*/README.md}}": [
+		"{docs/{root-manifest.json,tool/*.js},packages/{*/README.md,*/src/store/{actions,selectors}.js,components/src/*/README.md}}": [
```
Comment (gziolo) [inline package.json]: I noticed an error in this expression.

File: packages/jest-preset-default/jest-preset.json [hunk 2]
```diff
@@ -21,6 +21,11 @@
 		"**/?(*.)(spec|test).js",
 		"**/test/*.js"
 	],
+	"testPathIgnorePatterns": [
+		"/node_modules/",
+		"<rootDir>/.*/build/",
```
Comment (gziolo) [inline packages/jest-preset-default/jest-preset.json]: Somehow I had test files copied over to build folders causing the same test fail triple times ...

I couldn't reproduce it later, however, I added ignore patterns to ensure it doesn't affect test runner.

**Comment by gziolo** [comment]
I contemplated a bit about plugins a bit more, it’s super hard to reason as we have only one editor instance at the moment. In the future, we can have `edit-post` and `edit-template` pages. In that case, Yoast SEO tools or Jetpack features like Publicize might not be applicable when being in the template context. This means we should have two concurrent plugins `editPost.PluginSidebar` and `editTemplate.PluginSidebar`. The same would apply to `editPost.PluginModal` and `editTemplate.PluginModal`.

@atimmer @omarreiss @gwwar @tyxla any thoughts on this from your perspective? I'm trying to find the best approach to expose those plugin components to simplify their usage. See also related PR: #8958.

**Comment by atimmer** [comment]
@gziolo This sounds good to me.

**Review by tyxla — COMMENTED** [review]
The suggested approach makes sense to me - as long as `edit-post` makes sense to have `@wordpress/plugins` as a dependency (which it does for me).

**Comment by gwwar** [comment]
@gziolo that sounds reasonable to me!

> I contemplated a bit about plugins a bit more, it’s super hard to reason as we have only one editor instance at the moment.

Would it help if we crafted a second editor instance in an exploratory PR? Maybe if we stub out a simple `edit-template` iteration, and see how working with the plugin points feels like .

**Comment by gziolo** [comment]
Closing this one as it got stale. I think we can get back to it when we start working on phase 2 and will have more than one page where plugins can be pinned. /cc @youknowriad

**Pull request closed without merging**